experimental/manual/oss_fuzz_vuln_prompt.py

```diff
@@ -0,0 +1,198 @@
+Perform the following tasks to identify and remediate the security vulnerability:
+
+1. Vulnerability Identification:
+--- Stack Trace ---
+{stacktrace}
+
+--- Changeset (Git Diff Format) ---
+{changeset_diff}
+
+--- Source Code ---
+{source_code_content}
+
+---
+* Carefully review the stack trace to identify the precise line and function where the crash occurs.
+* Determine if the changeset directly introduced the vulnerability by correlating the crash stace with the changeset.
+* Exclude: Do not analyze the fuzzer code, test files, or build scripts for vulnerabilities.
+* If the changeset is not the cause, thoroughly examine the provided source code files to identify the root cause of the vulnerability.
+* Provide a clear description of the vulnerability, including:
+  * Type: Specify the vulnerability category (e.g., buffer overflow, SQL injection, cross-site scripting).
+  * Impact: Describe the potential consequences of exploitation (e.g., code execution, data loss, unauthorized access).
+  * Code Patterns: Explain the specific code constructs or patterns that allow the vulnerability to exist.
+* Approximate line number matches when comparing changeset to stack trace, as changeset revision is typically earlier.
+
+2. Technical Explanation:
+* Provide a detailed, step-by-step technical explanation of how an attacker could exploit the identified vulnerability.
+* Describe the code's logic flow, emphasizing the conditions or inputs that trigger the vulnerability.
+* Use diagrams, flowcharts, or annotated code snippets to illustrate the vulnerability's execution path and data flow.
+
+3. Remediation Patch:
+* Create a patch that completely addresses the root cause of the vulnerability.
+* Return the patch in git diff format (Use prefix '+' for added and '-' for removed lines).
+* Ensure the patch applies cleanly on the source code files provided.
+* Ensure the patch preserves the intended functionality of the codebase without introducing regressions.
+* Thoroughly test the patch in the relevant environment to verify that it eliminates the crash and prevents exploitation.
+"""
+PROMPT_MAX_LENGTH = 1048576
+PROJECTS_DIR = os.path.join('oss-fuzz', 'projects')
+STACK_FRAME_START_REGEX = re.compile(r'\s*#\d+\s+0x[0-9A-Fa-f]+\s+')
+STACK_FRAME_PATH_LINE_REGEX = re.compile(
+    r'(?<=\[|\(|\s)([a-zA-Z/.][^\s]*?)\s*(:|@)\s*(\d+)(?=\]$|\)$|:\d+$|$)')
+EXCLUDED_FILE_PATH_SUBSTRINGS = ('/compiler-rt/', '/glibc-')
+
+
+def get_local_repo_path(repo_url):
+  """Returns the local path of the repository."""
+  local_repo_name = repo_url.split("/")[-1]
+  return os.path.join(PROJECTS_DIR, local_repo_name)
+
+
+def get_commit_range(regression_range):
+  """
+    Parses a regression range string and returns the start and end commits.
+
+    Args:
+        regression_range: A string representing the regression range, in formats like:
+            - "start_commit:end_commit"
+            - "start_commit..end_commit"
+            - "end_commit" 
```

Comment:
Just out of curiosity, if we can assume these are the only 3 formats, could we also assume the length of each commit?
If so, would it be too hacky to do:
```python
if len(regression) > N:
  return regression_range[N:], regression_range[-N:]
return f'regression_range~', regression_range
```

Reply:
No too hacky, sticking with more readable and correct code.